constants.py

```diff
@@ -0,0 +1,18 @@
+""" In this file there are constant global variables used in build_wheels.py script """
+# URL to fetch IDF branches from
+IDF_BRANCHES_URL = 'https://api.github.com/repos/espressif/esp-idf/branches?protected=true&per_page=100'
+
+# Minimal IDF release version to take requirements from (v{MAJOR}.{MINOR})
+# Requirements from all release branches and master equal or above this will be considered
+MIN_IDF_MAJOR_VERSION: int = 5
+MIN_IDF_MINOR_VERSION: int = 0
+
+# Constraints file version is automatically taken from branch name
+# For master branch it is taken constraints file incremented by 1 in IDF minor version
+# (e.g when last release is v5.2 the constrains file for master is v5.3)
+# When no such a file exists the last release constraints file is taken for master branch
+#
+# This feature can be overwritten by setting the AUTO_VERSION_CONSTRAINS_FILE to False
+# then the value in MASTER_CONSTRAIN_FILE is taken
+AUTO_VERSION_CONSTRAINS_FILE: bool = True
```

Comment:
Thank you @tomassebestik, if I am not wrong `AUTO_VERSION_CONSTRAINS_FILE=false` will be needed if last `release` version of IDF will be `v5.X` and `master` will be planed to have major release to `v6.0`, the auto-version mechanism for constraints file will be assigning the `master` branch `v5.(X+1)` constraints file which will not exist and than it falls back to last release constraints file version for `master`.

So in that case I thought it might be useful to have "force overwrite" option of constraints file for `master` branch.

Reply:
But after releasing v6.1, one also has to go and set back AUTO_VERSION_CONSTRAINS_FILE to true.

I don't like the fact that v6.0 is hard-coded in two different files as well. We will also have to remember that those need to be set to v7.0 as well?

Instead of having such a complicated flow why couldn't we just download https://github.com/espressif/esp-idf/blob/master/tools/cmake/version.cmake (without cloning) and get the version from it? We wouldn't have to make any modification in the future.